### DISCUSSION TITLE
.parcel-cache cleanup recommendations

### DISCUSSION THREAD

**Discussion by gregid** [parcel-bundler/parcel#5048] [General]
As there is an ongoing issue with cache causing some weird bugs I am trying to collect instances where `.parcel-cache` cleanup helped you fix the problem.

In my case these were:
- after `npm install` including parcel update
- with changes to local package installed with `npm link`

any other cases you know of?

**Comment by mischnic** [comment]
> with changes to local package installed with npm link

This sounds like https://github.com/parcel-bundler/parcel/issues/4950.

> after npm install including parcel update

Probably https://github.com/parcel-bundler/parcel/issues/5032, first half will be fixed in https://github.com/parcel-bundler/parcel/pull/4997

**Reply by gregid** [reply]
@mischnic good to know cache invalidation for npm install is being worked on.
Since it looks like symlinks may take a while to be resolved how do you guys deal with local packages that works with current parcel2?
I was thinking of trying https://johanronsse.be/2020/08/17/using-yalc-as-an-alternative-to-npm-link/ but haven't tested it yet.
Deleting cache after every change to symlinked package is quite painful.

**Reply by mischnic** [reply]
I agree, we either need to add this functionality to the watcher or always invalidate symlinked folders outside the project root.
